File: backend/auth/src/main/java/com/watson/auth/realtor/domain/entity/Realtor.java
package com.watson.auth.realtor.domain.entity;

import com.watson.auth.util.BaseEntity;
import javax.persistence.*;
import lombok.*;
import lombok.experimental.SuperBuilder;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder
@Data
@Table(name="realtors")
public class Realtor extends BaseEntity {

    @Id
    private String id;

    // 중개사 정보
    @Column(nullable = false)
    private String license;

    @Column(nullable = false)
    private String realtorName;

    @Column(nullable = false)
    private String phoneNumber;

    // 중개사무소 정보
    @Column(nullable = false)
    private String registId;

    @Column(nullable = false)
    private String agencyName;

    @Column(nullable = false)
    private String address;

    @Column(nullable = false)
    private String tel;

    @Column(nullable = false)
    private String agencyImg;

}

File: frontend/prac2/lib/user/models/realtor.dart
// 공인중개사 데이터 모델
class Realtor {
  final String license;
  final String realtorName;
  final int phoneNumber;
  final String authId;
  final int registId;
  final String address;
  final int? tel;

  Realtor({
    required this.license,
    required this.authId,
    required this.realtorName,
    required this.phoneNumber,
    required this.registId,
    required this.address,
    required this.tel,
  });

  factory Realtor.fromJson(Map<String, dynamic> json) {
    return Realtor(
      license: json['license'],
      authId: json['authId'],
      realtorName: json['realtorName'],
      phoneNumber: json['phoneNumber'],
      registId: json['registId'],
      address: json['address'],
      tel: json['tel'],
    );
  }
}